src/components/data-exploration/generic-gene-table/GeneTable.jsx

```diff
@@ -42,6 +43,7 @@ const GeneTable = (props) => {
 
   useEffect(() => {
```

Comment:
What about having a custom useEffect hook that handles only the matomo  tracking? To split it up from the rest of the page logic.